### PR TITLE
DAOS-6407: list_pools.py weekly test is failing

### PR DESCRIPTION
**Pull request by cibervicho**
This test was updated on a previous commit and the number of ranks used to
iterate over it was not limited to the required one as the test was expecting.

Also fixed the regular expression used by the 'def pool_list()' method since, with
the recent updates to dmg util, the result of 'dmg pool list' command was not
getting any match in the cases of multiple service replicas shown like '[1-3]'.

Signed-off-by: David Maldonado Moreno <david.maldonado.moreno@intel.com>

### CODE REVIEW

**Comment by cibervicho** [comment]
Test passed in both, CentOS 7 and Leap 15, Functional steps:
`2021-01-20 17:49:01,590 test             L0779 INFO | PASS 1-./pool/list_pools.py:ListPoolsTest.test_list_pools;hosts-servers-d09b

2021-01-20 17:53:25,157 test             L0779 INFO | PASS 1-./pool/list_pools.py:ListPoolsTest.test_list_pools;hosts-servers-ae79`

**Review by shimizukko — DISMISSED** [review]
LGTM

File: src/tests/ftest/pool/list_pools.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ def test_list_pools(self):
             ),
             (
                 "Create 3 pools using all ranks with --svcn=3",
```
Comment (berserk-fury) [inline src/tests/ftest/pool/list_pools.py:115]: should it be --nsvc instead of --svcn?

**Review by berserk-fury — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]
LGTM

**Comment by cibervicho** [comment]
Dear @daos-stack/daos-gatekeeper, could you help with this PR?

**Pull request merged**